src/BoundaryConditions/update_boundary_conditions.jl

```diff
@@ -0,0 +1,16 @@
+using Oceananigans: boundary_conditions
+
+update_boundary_condition!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
+
+function update_boundary_condition!(bcs::FieldBoundaryConditions, field, model)
```

Comment:
As noted before, if we can pass less information than the full `model` it may help.

Reply:
any progress on this?

Reply:
the main downside being, as someone reading the code I don't know what parts of `model` are actually needed to `update_boundary_condition!`

Reply:
Sorry, I missed this one, I'm not sure what someone might need in it. Model has:
```
advection           architecture        auxiliary_fields    biogeochemistry
buoyancy            clock               closure             coriolis
diffusivity_fields  forcing             free_surface        grid
particles           pressure            timestepper         tracers
velocities
```
I guess we'll never need or `biogeochemistry`, `forcing` (probably?), `particles`, or `timestepper`, but I could see the others being useful for complicated boundary computations? In the same way `update_state!` could do loads of different things it might be safer to keep this generic for now so we don't have to break it if someone wants todo something different with it in the future.

In the case, I had in mind (updating the mean wall normal velocity) we only need velocities from the model but other things that try to conserve some properties on the boundary might need most/all of the things effecting the dynamics.